src/app/component/user/components/profile/shopping-list/shopping-list.component.ts

```diff
@@ -22,9 +22,13 @@ export class ShoppingListComponent implements OnInit, OnDestroy {
   public getShoppingList(): void {
     this.profileSubscription = this.profileService.getShoppingList()
     .subscribe(
-      (shoppingListArr: ShoppingList[]) => this.shoppingList = shoppingListArr,
-      error => this.shoppingList = []
+      (shoppingListArr: ShoppingList[]) => this.shoppingList = shoppingListArr
     );
+    this.getLength();
+  }
+
+  public getLength() {
+    return this.shoppingList.length;
```

Comment:
what if this.shoppingList will be undefined?

Reply:
Done